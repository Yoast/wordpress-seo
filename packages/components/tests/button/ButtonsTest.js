/* eslint-disable react/jsx-no-bind */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import {
	NewButton as Button,
	ButtonStyledLink,
	CloseButton,
} from "../../src/button";

const shallowRenderer = new ReactShallowRenderer();
describe( "Button", () => {
	it( "should render with only required props", () => {
		shallowRenderer.render( <Button /> );

		const result = shallowRenderer.getRenderOutput();

		expect( result ).toBeDefined();
	} );

	it( "should render a button based on provided props", () => {
		shallowRenderer.render(
			<Button
				onClick={ ()=>{} }
				id="very-nice-id"
			/>
		);

		const result = shallowRenderer.getRenderOutput();

		expect( result ).toBeDefined();
		expect( result.props.id ).toBe( "very-nice-id" );
		expect( result.props.className ).toBe( "yoast-button yoast-button--primary" );
		expect( result.type ).toBe( "button" );
	} );

	it( "should render a small button based on the provided 'small' prop", () => {
		shallowRenderer.render(
			<Button
				onClick={ ()=>{} }
				id="very-nice-id"
				variant="buy"
				small={ true }
			/>
		);

		const result = shallowRenderer.getRenderOutput();

		expect( result ).toBeDefined();
		expect( result.props.className ).toBe( "yoast-button yoast-button--buy yoast-button--small" );
	} );

	it( "should render a small button based on the provided 'disabled' prop", () => {
		shallowRenderer.render(
			<Button
				onClick={ ()=>{} }
				id="very-nice-id"
				variant="secondary"
				disabled={ true }
			/>
		);

		const result = shallowRenderer.getRenderOutput();

		expect( result ).toBeDefined();
		expect( result.props.className ).toBe( "yoast-button yoast-button--secondary" );
	} );

	it( "generates an error if wrong props are provided", () => {
		console.error = jest.fn();

		shallowRenderer.render(
			<Button
				variant={ { afterIcon: "blerf" } }
			/>
		);

		expect( console.error ).toBeCalled();
		expect( console.error.mock.calls[ 0 ][ 0 ] )
			.toContain( "Warning: Failed prop type: Invalid prop `variant` of value `[object Object]` supplied to `Button`" );
	} );
} );

describe( "Primary variant", () => {
	const expectedClassName = "yoast-button yoast-button--primary";
	it( "matches the Button snapshot", () => {
		shallowRenderer.render( <Button variant="primary" onClick={ () => {} }>test</Button> );

		const result = shallowRenderer.getRenderOutput();
		expect( result ).toMatchSnapshot();
		expect( result.props.className ).toBe( expectedClassName );
	} );

	it( "matches the Link snapshot", () => {
		shallowRenderer.render( <ButtonStyledLink variant="primary" href="#">test</ButtonStyledLink> );

		const result = shallowRenderer.getRenderOutput();
		expect( result ).toMatchSnapshot();
		expect( result.props.className ).toBe( expectedClassName );
	} );
} );

describe( "Secondary variant", () => {
	const expectedClassName = "yoast-button yoast-button--secondary";
	it( "matches the Button snapshot", () => {
		shallowRenderer.render( <Button variant="secondary" onClick={ () => {} }>test</Button> );

		const result = shallowRenderer.getRenderOutput();
		expect( result ).toMatchSnapshot();
		expect( result.props.className ).toBe( expectedClassName );
	} );

	it( "matches the Link snapshot", () => {
		shallowRenderer.render( <ButtonStyledLink variant="secondary" href="#">test</ButtonStyledLink> );

		const result = shallowRenderer.getRenderOutput();
		expect( result ).toMatchSnapshot();
		expect( result.props.className ).toBe( expectedClassName );
	} );
} );

describe( "Buy variant", () => {
	const expectedClassName = "yoast-button yoast-button--buy";
	it( "matches the Button snapshot", () => {
		shallowRenderer.render( <Button variant="buy" onClick={ () => {} }>test</Button> );

		const result = shallowRenderer.getRenderOutput();
		expect( result ).toMatchSnapshot();
		expect( result.props.className ).toBe( expectedClassName );
	} );

	it( "matches the Link snapshot", () => {
		shallowRenderer.render( <ButtonStyledLink variant="buy" href="#">test</ButtonStyledLink> );

		const result = shallowRenderer.getRenderOutput();
		expect( result ).toMatchSnapshot();
		expect( result.props.className ).toBe( expectedClassName );
	} );
} );

describe( "Text variant", () => {
	const removeClassName = "yoast-remove";
	it( "matches the Button snapshot", () => {
		shallowRenderer.render( <Button variant="remove" onClick={ () => {} }>test</Button> );

		const result = shallowRenderer.getRenderOutput();
		expect( result ).toMatchSnapshot();
		expect( result.props.className ).toBe( removeClassName );
	} );

	it( "matches the Link snapshot", () => {
		shallowRenderer.render( <ButtonStyledLink variant="remove" href="#">test</ButtonStyledLink> );

		const result = shallowRenderer.getRenderOutput();
		expect( result ).toMatchSnapshot();
		expect( result.props.className ).toBe( removeClassName );
	} );

	const hideClassName = "yoast-hide";
	it( "matches the Button snapshot", () => {
		shallowRenderer.render( <Button variant="hide" onClick={ () => {} }>test</Button> );

		const result = shallowRenderer.getRenderOutput();
		expect( result ).toMatchSnapshot();
		expect( result.props.className ).toBe( hideClassName );
	} );

	it( "matches the Link snapshot", () => {
		shallowRenderer.render( <ButtonStyledLink variant="hide" href="#">test</ButtonStyledLink> );

		const result = shallowRenderer.getRenderOutput();
		expect( result ).toMatchSnapshot();
		expect( result.props.className ).toBe( hideClassName );
	} );
} );

describe( "Icon", () => {
	const closeClassname = "yoast-close";
	it( "CloseButton matches the Button snapshot", () => {
		shallowRenderer.render( <CloseButton onClick={ () => {} }>test</CloseButton> );

		const result = shallowRenderer.getRenderOutput();
		expect( result ).toMatchSnapshot();
		expect( result.props.className ).toBe( closeClassname );
	} );
} );
