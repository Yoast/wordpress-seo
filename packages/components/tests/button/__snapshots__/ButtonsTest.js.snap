// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buy variant matches the Button snapshot 1`] = `
<button
  className="yoast-button yoast-button--buy"
  onClick={[Function]}
  type="button"
>
  test
  <span
    className="yoast-button--buy__caret"
  />
</button>
`;

exports[`Buy variant matches the Link snapshot 1`] = `
<a
  className="yoast-button yoast-button--buy"
  href="#"
>
  test
  <span
    className="yoast-button--buy__caret"
  />
</a>
`;

exports[`Icon CloseButton matches the Button snapshot 1`] = `
<button
  aria-label="Close"
  className="yoast-close"
  onClick={[Function]}
>
  <svg
    aria-hidden="true"
    focusable="false"
    role="img"
    viewBox="0 0 352 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
    />
  </svg>
</button>
`;

exports[`Primary variant matches the Button snapshot 1`] = `
<button
  className="yoast-button yoast-button--primary"
  onClick={[Function]}
  type="button"
>
  test
</button>
`;

exports[`Primary variant matches the Link snapshot 1`] = `
<a
  className="yoast-button yoast-button--primary"
  href="#"
>
  test
</a>
`;

exports[`Secondary variant matches the Button snapshot 1`] = `
<button
  className="yoast-button yoast-button--secondary"
  onClick={[Function]}
  type="button"
>
  test
</button>
`;

exports[`Secondary variant matches the Link snapshot 1`] = `
<a
  className="yoast-button yoast-button--secondary"
  href="#"
>
  test
</a>
`;

exports[`Text variant matches the Button snapshot 1`] = `
<button
  className="yoast-remove"
  onClick={[Function]}
  type="button"
>
  test
</button>
`;

exports[`Text variant matches the Button snapshot 2`] = `
<button
  className="yoast-hide"
  onClick={[Function]}
  type="button"
>
  test
</button>
`;

exports[`Text variant matches the Link snapshot 1`] = `
<a
  className="yoast-remove"
  href="#"
>
  test
</a>
`;

exports[`Text variant matches the Link snapshot 2`] = `
<a
  className="yoast-hide"
  href="#"
>
  test
</a>
`;
